subsys/ipc/ipc_service/lib/ipc_static_vrings.c

```diff
@@ -175,7 +142,20 @@ int ipc_static_vrings_init(struct ipc_static_vrings *vr, unsigned int role)
 	metal_io_init(vr->shm_device.regions, (void *) vr->shm_addr,
 		      vr->shm_physmap, vr->shm_size, -1, 0, NULL);
 
-	err = libmetal_setup(vr);
+	err = metal_register_generic_device(&vr->shm_device);
+	if (err != 0) {
+		return err;
+	}
+
+	err = metal_device_open("generic", vr->shm_device.name, &device);
+	if (err != 0) {
+		return err;
+	}
+
+	vr->shm_io = metal_device_io_region(device, 0);
+	if (vr->shm_io == NULL) {
+		return err;
+	}
```

Comment:
Same as above, goes away in the end. I guess if we flipped the patch order and removed metal_device first it would be fixed at that point. Problem was libmetal_setup ends up being a one line function after removing metal_device, I'd be way too tempted to inline that as part of the metal_device removal patch.

Maybe that would work, I could just squash the two patches together, they are really just two steps in the same logical change.. up to you.

Reply:
On second thought, flipping the order does make it clean enough to be worth it, will flip them.